pkg/scheduler/scheduler.go

```diff
@@ -458,10 +453,11 @@ func (s *Scheduler) scheduleResourceBindingWithClusterAffinity(rb *workv1alpha2.
 	klog.V(4).Infof("ResourceBinding(%s/%s) scheduled to clusters %v", rb.Namespace, rb.Name, scheduleResult.SuggestedClusters)
 	patchErr := s.patchScheduleResultForResourceBinding(rb, string(placementBytes), scheduleResult.SuggestedClusters)
```

Comment:
What if aggregate patchErr and error first, and return this aggregation one?

Reply:
we do not care about what type of error to retry scheduling, but for `NoClusterFit`, we cannot judge `FitError` by `errors.As` function if the aggregated error for `patchErr` and `fitErr` is returnd.

Reply:
```go
patchErr := s.patchScheduleResultForResourceBinding(rb, string(placementBytes), scheduleResult.SuggestedClusters)
err = utilerrors.NewAggregate([]error{err, patchErr})
s.recordScheduleResultEventForResourceBinding(rb, err)
return err
```